precompiles/staking/staking.go

```diff
@@ -231,6 +232,13 @@ func (c *Contract) Stake(
 		return nil, err
 	}
 
+	// if caller is not the same as origin it means call is coming through smart contract
+	// in that case, manually reduce amount in stateDB so it is properly reflected in bank module
```

Comment:
Maybe we can give more details in the comment why is it necessary when the caller is not origin